src/util.ts

```diff
@@ -38,12 +38,72 @@ export function nop() {
 /**
  * Find the tsconfig.json, read it, and return parsed contents.
  * @param rootDir Directory where the tsconfig.json should be found.
+ * If the tsconfig.json file has an "extends" field hop down the dependency tree
+ * until it ends or a circular reference is found in which case an error will be
+ * thrown
  */
 export async function getTSConfig(
-    rootDir: string, customReadFilep?: ReadFileP): Promise<{}> {
-  const tsconfigPath = path.join(rootDir, 'tsconfig.json');
+    rootDir: string, customReadFilep?: ReadFileP): Promise<ConfigFile> {
   customReadFilep = customReadFilep || readFilep;
-  const json = await customReadFilep(tsconfigPath, 'utf8');
-  const contents = JSON.parse(json);
+  const readArr = new Set();
+  return await getBase('tsconfig.json', customReadFilep, readArr, rootDir);
+}
+
+/**
+ * Recursively iterate through the dependency chain until we reach the end of
+ * the dependency chain or encounter a circular reference
+ * @param filePath Filepath of file currently being read
+ * @param customReadFilep The file reading function being used
+ * @param readFiles an array of the previously read files so we can check for
+ * circular references
+ * returns a ConfigFile object containing the data from all the dependencies
+ */
+async function getBase(
+    filePath: string, customReadFilep: ReadFileP, readFiles: Set<string>,
+    currentDir: string): Promise<ConfigFile> {
+  customReadFilep = customReadFilep || readFilep;
+
+  filePath = path.resolve(currentDir, filePath);
+
+  if (readFiles.has(filePath)) {
+    throw new Error('Circular reference in ${filePath}');
```

Comment:
For posterity, can you include a comment on why we throw here (rather than give up and return immediately)?